### ISSUE TITLE
Np, tf and torch initialize arrays with different default dtype

### ISSUE THREAD

**Issue by ninamiolane** [geomstats/geomstats#775]
The following code:
```
import numpy as np
import tensorflow as tf
import torch

a = np.array([1.])
b = tf.constant([1.])
c = torch.tensor([1.])
print(a.dtype, b.dtype, c.dtype)
```
gives:
```
float64 torch.float32 <dtype: 'float32'>
```

Similarly:
```
import geomstats.backend as gs

a = gs.array([1.])
print(a.dtype)
```
outputs float64, float32, float32 with np, tf and torch respectively.

Do we want to uniformize this and if so, use float64 or float32 as a default?

**Comment by nguigs** [comment]
That's probably why we always have to adapt tol for tests in pytorch and tensorflow! I would have said float32 but we still need to gain on numerical accuracy...

**Comment by ninamiolane** [comment]
We could have a `DEFAULT_FLOAT_DTYPE = ...` in each backend, and initially set it to `gs.float32` (?). We could revert it if `gs.float64` seems more convenient?

**Comment by lpereira95** [comment]
Unless it has changed in recent years (which I don't believe), `numpy` does not have a way of setting `float32` as default (see [this](https://github.com/numpy/numpy/issues/6860)).

So in my opinion we should set `float64` as default for all backends (for consistency) and let the user decide to reduce precision if it applies to his/her use case (it will be specially relevant when using gpu).

**Comment by lpereira95** [comment]
Solved by #1615.
